src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseCancellationTokenThrowIfCancellationRequested.Fixer.cs

```diff
@@ -0,0 +1,108 @@
+            IOperation? whenFalse = UseCancellationTokenThrowIfCancellationRequested.GetSingleStatementOrDefault(conditional.WhenFalse);
+
+            Func<CancellationToken, Task<Document>> createChangedDocument;
+            if (symbols.IsSimpleAffirmativeCheck(conditional, out IPropertyReferenceOperation? propertyReference))
+            {
+                //  For simple checks of the form:
+                //      if (token.IsCancellationRequested)
+                //          throw new OperationCanceledException();
+                //  Replace with:
+                //      token.ThrowIfCancellationRequested();
+                createChangedDocument = async token =>
+                {
+                    var editor = await DocumentEditor.CreateAsync(context.Document, token).ConfigureAwait(false);
+                    SyntaxNode expressionStatement = CreateThrowIfCancellationRequestedExpressionStatement(editor, conditional, propertyReference);
+                    editor.ReplaceNode(conditional.Syntax, expressionStatement);
+                    return editor.GetChangedDocument();
+                };
+            }
+            else if (symbols.IsNegatedCheckWithThrowingElseClause(conditional, out propertyReference))
+            {
+                //  For negated checks of the form:
+                //      if (!token.IsCancellationRequested) { DoStatements(); }
+                //      else { throw new OperationCanceledException(); }
+                //  Replace with:
+                //      token.ThrowIfCancellationRequested();
+                //      DoStatements();
+                createChangedDocument = async token =>
+                {
+                    var editor = await DocumentEditor.CreateAsync(context.Document, token).ConfigureAwait(false);
+                    SyntaxNode expressionStatement = CreateThrowIfCancellationRequestedExpressionStatement(editor, conditional, propertyReference);
+                    editor.ReplaceNode(conditional.Syntax, expressionStatement);
+                    if (conditional.WhenTrue is IBlockOperation block)
+                    {
+                        editor.InsertAfter(expressionStatement, block.Operations.Select(x => x.Syntax.WithAdditionalAnnotations(Formatter.Annotation)));
+                    }
+                    else
+                    {
+                        editor.InsertAfter(expressionStatement, conditional.WhenTrue.Syntax);
+                    }
+
+                    //  Ensure if-blocks with multiple statements maintain correct indentation.
+                    return await Formatter.FormatAsync(editor.GetChangedDocument(), Formatter.Annotation, cancellationToken: token).ConfigureAwait(false);
+                };
+            }
+            else
+            {
+                return;
+            }
+
+            var codeAction = CodeAction.Create(
+                Resx.UseCancellationTokenThrowIfCancellationRequestedTitle,
```

Comment:
@mavasani The title makes sense as code-fix title in this case, would you prefer to have a separate entry for it anyway (with the same content).

Reply:
I've added a separate resource for the code fix title to be consistent.